mteb/abstasks/mteb_result.py

```diff
@@ -0,0 +1,182 @@
+        for split, hf_subset_scores in v.items():
+            for lang, scores in hf_subset_scores.items():
+                cls._validate_scores_dict(scores)
+        return v
+
+    @staticmethod
+    def _validate_scores_dict(scores: Scores) -> None:
+        if "evaluation_time" not in scores:
+            raise ValueError("'evaluation_time' should be in scores")
+
+        if "main_score" not in scores:
+            raise ValueError("'main_score' should be in scores")
+
+        # check that it is json serializable
+        try:
+            _ = json.dumps(scores)
+        except Exception as e:
+            raise ValueError(f"Scores are not json serializable: {e}")
+
+    def to_dict(self) -> dict:
+        return self.model_dump()
+
+    @classmethod
+    def from_dict(cls, data: dict) -> MTEBResults:
+        return cls.model_validate(data)
+
+    def to_disk(self, path: Path) -> None:
+        with path.open("w") as f:
+            f.write(self.model_dump_json())
+
+    @classmethod
+    def from_disk(cls, path: Path) -> MTEBResults:
+        with path.open("r") as f:
+            return cls.model_validate(json.load(f))
+
+    def get_score(
+        self,
+        getter: Callable[[Scores], float],
+        aggregation: Callable[[list[float]], Any],
+        splits: list[Split] | None = None,
+        languages: list[ISO_LANGUAGE | ISO_LANGUAGE_SCRIPT] | None = None,
+        scripts: list[ISO_LANGUAGE_SCRIPT] | None = None,
+    ) -> Any:
+        """Get a score for the specified splits, languages, scripts and aggregation function.
+
+        Args:
+            getter: A function that takes a scores dictionary and returns a score e.g. "main_score" or "evaluation_time".
+            aggregation: The aggregation function to use.
+            splits: The splits to consider.
+            languages: The languages to consider. Can be ISO language codes or ISO language script codes.
+            scripts: The scripts to consider.
+
+        Returns:
+            The result of the aggregation function on the scores.
+        """
+        meta = TASKS_REGISTRY[self.task_name].metadata
+        hf_subset2langs = eval_langs_as_dict(meta.eval_langs)
+
+        if splits is None:
+            splits = list(self.scores.keys())
+
+        lang_script_filter = LangScriptFilter.from_languages_and_scripts(
+            languages, scripts
+        )
+
+        values = []
+        for split in splits:
+            if split not in self.scores:
+                raise ValueError(f"Split {split} not found in scores")
+
+            for hf_subset, scores in self.scores[split].items():
+                eval_langs = hf_subset2langs[hf_subset]
+                include_subset = False
+                for lang in eval_langs:
+                    if lang_script_filter.is_valid_language(lang):
+                        include_subset = True
+                        break
+                if include_subset:
+                    values.append(getter(scores))
```

Comment:
Could be simplified?
```suggestion
                    if lang_script_filter.is_valid_language(lang):
                        values.append(getter(scores))
                        break
```

Reply:
Agree. Added this mostly to clarify what the for loops check for.

Reply:
fixed